jobqueue.go

```diff
@@ -113,8 +131,15 @@ func (s *Stack) run() {
 		case <-timeout:
 			oldest.notify <- ErrTimeout
 			s.stack.shift()
+		case statusRequest := <-s.status:
+			statusRequest <- Status{Active: s.busy, Queued: s.stack.list.Len()}
 		case <-s.quit:
-			// TODO: teardown
+			for !s.stack.empty() {
+				j := s.stack.shift()
+				j.notify <- ErrClosed
```

Comment:
this is a force quit instead of graceful shutdown

Reply:
Let's document the behavior and let it up to the user of this library if and when to call the force shutdown.
If we have visibility of the queue active jobs and queue size, we are fine from the user pov.

Reply:
changed the Close() such that it waits for all jobs, but doesn't accept new jobs. Also added CloseForced() variant, that is equivalent with the previous approach.